core/storage/suggestion/gae_models.py

```diff
@@ -821,7 +825,18 @@ def get_in_review_question_suggestions_by_offset(
                     break
                 for suggestion_model in suggestion_models:
                     offset += 1
-                    if suggestion_model.author_id != user_id:
+
+                    included_in_topics = False
+                    if topics is not None:
+                        for topic in topics:
+                            topic_skills = topic.get_all_skill_ids()
+
+                            if suggestion_model.target_id in topic_skills:
+                                included_in_topics = True
+                                break
```

Comment:
The storage model should only make storage queries and not call other backend services. I think the controller (suggestion.py) should compute all the skill IDs that are part of the selected topic and then we modify the backend services and storage query to accept the list of skill IDs and use the skill ID list as the filter here.

Reply:
Good idea, will do!